permissions/resource_rule_set.go

```diff
@@ -0,0 +1,177 @@
+			Etag: "",
+		})
+		if err != nil {
+			return "", err
+		}
+		ruleSetUpdateReq.RuleSet.Etag = ruleSetGetRes.Etag
+		ruleSetUpdateRes, err := rsApi.Update(ruleSetUpdateReq)
+		if err != nil {
+			return "", err
+		}
+		return ruleSetUpdateRes.Etag, nil
+	}
+	handleConflictAndUpdate := func(ruleSetUpdateReq iam.UpdateRuleSetRequest, rsApi RuleSetApi) (string, error) {
+		ruleSetUpdateRes, err := rsApi.Update(ruleSetUpdateReq)
+		if err != nil {
+			if strings.EqualFold(err.Error(), "Conflict with another RuleSet operation") {
+				// we need to get and update
+				etag, err := getAndUpdateRuleSet(ruleSetUpdateReq, rsApi)
+				return etag, err
+			}
+			return "", err
+		}
+		return ruleSetUpdateRes.Etag, err
+	}
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var ruleSetUpdateReq iam.UpdateRuleSetRequest
+			common.DataToStructPointer(d, s, &ruleSetUpdateReq.RuleSet)
+			ruleSetUpdateReq.Name = ruleSetUpdateReq.RuleSet.Name
+			rsApi := NewRuleSetApi(ctx, c)
+			etag, err := getAndUpdateRuleSet(ruleSetUpdateReq, rsApi)
+			if err != nil {
+				return err
+			}
+			d.Set("etag", etag)
+			d.SetId(ruleSetUpdateReq.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			name := d.Id()
+			rsApi := NewRuleSetApi(ctx, c)
+			getRuleSetReq := iam.GetRuleSetRequest{
+				Name: name,
+				Etag: "",
+			}
+			if etag, ok := d.GetOk("etag"); ok {
+				getRuleSetReq = iam.GetRuleSetRequest{
+					Name: name,
+					Etag: fmt.Sprintf("%v", etag),
+				}
+			}
+			data, err := rsApi.Read(getRuleSetReq)
+			if err != nil {
+				return err
+			}
+			return common.StructToData(data, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var ruleSetUpdateReq iam.UpdateRuleSetRequest
+			common.DataToStructPointer(d, s, &ruleSetUpdateReq.RuleSet)
+			ruleSetUpdateReq.Name = ruleSetUpdateReq.RuleSet.Name
+			rsApi := NewRuleSetApi(ctx, c)
+			if etag, ok := d.GetOk("etag"); ok {
+				ruleSetUpdateReq.RuleSet.Etag = fmt.Sprintf("%v", etag)
+				updatedEtag, err := handleConflictAndUpdate(ruleSetUpdateReq, rsApi)
+				if err != nil {
+					return err
+				}
+				d.Set("etag", updatedEtag)
+				return nil
+			}
+			etag, err := getAndUpdateRuleSet(ruleSetUpdateReq, rsApi)
+			if err != nil {
+				return err
+			}
+			d.Set("etag", etag)
+			return nil
+		},
+		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			// we remove all grant rules. Account admins will still be able to update rule set
+			name := d.Id()
+			rsApi := NewRuleSetApi(ctx, c)
+			updateRuleSetReq := iam.UpdateRuleSetRequest{
+				Name: name,
+				RuleSet: iam.RuleSetUpdateRequest{
+					Name: name,
+					Etag: "",
+				},
+			}
+			if etag, ok := d.GetOk("etag"); ok {
+				updateRuleSetReq.RuleSet.Etag = fmt.Sprintf("%v", etag)
```

Comment:
type cast it to string:
```suggestion
				updateRuleSetReq.RuleSet.Etag = etag.(string)
```